include/deal.II/non_matching/mapping_info.h

```diff
@@ -45,6 +46,10 @@ namespace NonMatching
   class MappingInfo : public Subscriptor
```

Comment:
I don't remember what exactly we discussed here, but I am a bit concerned about having this in the user-visible namespace, as `MappingInfo` does not tell very much. I used it inside an internal namespace of `MatrixFree` because I did not want to spell it out as `MappingRelatedData` or maybe even find a different name, to avoid confusion with https://github.com/dealii/dealii/blob/7331b14d3e3d68033c9dc0b64631ce3b66db7bc2/include/deal.II/fe/mapping_related_data.h#L39-L52

In any case, I see the desire to have the same name as https://github.com/dealii/dealii/blob/7331b14d3e3d68033c9dc0b64631ce3b66db7bc2/include/deal.II/matrix_free/mapping_info.h#L48-L55, but in that case I suggest to be clear with the doxygen description of the class, using inspiration from the other two classes (I admit `MatrixFreeFunctions::MappingInfo` is also not good, but we might extend there as well).

Reply:
I agree that the current naming is not optimal but we should discuss where to put this separately from this PR. This is also connected to the question in which namespace and folder `FEPointEvaluation` belongs.